packages/e2e-tests/src/tests/userProfile.test.ts

```diff
@@ -0,0 +1,187 @@
+    const fullMessages = await generalChannelOwner.getUserMessagesFull(users.owner.username)
+    const img = await fullMessages[1].findElement(By.tagName('img'))
+    const imgSrc = await img.getAttribute('src')
+    expect(imgSrc).toEqual(expectedImgSrc)
+  })
+
+  it('Owner opens the settings tab and gets an invitation code', async () => {
+    const settingsModal = await new Sidebar(users.owner.app.driver).openSettings()
+    const isSettingsModal = await settingsModal.element.isDisplayed()
+    expect(isSettingsModal).toBeTruthy()
+    await sleep(2000)
+    await settingsModal.switchTab('invite') // TODO: Fix - the invite tab should be default for the owner
+    await sleep(2000)
+    const invitationCodeElement = await settingsModal.invitationCode()
+    await sleep(2000)
+    invitationCode = await invitationCodeElement.getText()
+    await sleep(2000)
+    expect(invitationCode).not.toBeUndefined()
+    log('Received invitation code:', invitationCode)
+    await settingsModal.close()
+  })
+
+  it('First user opens the app', async () => {
+    await users.user1.app.open()
+  })
+
+  it('First user submits invitation code received from owner', async () => {
+    const joinCommunityModal = new JoinCommunityModal(users.user1.app.driver)
+    const isJoinCommunityModal = await joinCommunityModal.element.isDisplayed()
+    expect(isJoinCommunityModal).toBeTruthy()
+    await joinCommunityModal.typeCommunityCode(invitationCode)
+    await joinCommunityModal.submit()
+  })
+
+  it('First user submits valid username', async () => {
+    const registerModal = new RegisterUsernameModal(users.user1.app.driver)
+    const isRegisterModal = await registerModal.element.isDisplayed()
+    expect(isRegisterModal).toBeTruthy()
+    await registerModal.clearInput()
+    await registerModal.typeUsername(users.user1.username)
+    await registerModal.submit()
+  })
+
+  it('First user joins successfully sees general channel', async () => {
+    generalChannelUser1 = new Channel(users.user1.app.driver, 'general')
+    await generalChannelUser1.element.isDisplayed()
+    const isMessageInput2 = await generalChannelUser1.messageInput.isDisplayed()
+    expect(isMessageInput2).toBeTruthy()
+  })
+
+  it("First user sees owner's message with profile photo", async () => {
+    const messages = await generalChannelUser1.getAtleastNumUserMessages(users.owner.username, 2)
+    const elem = messages?.[1]
+    if (!elem) {
+      fail('Failed to find at least 2 messages')
+    }
+    await users.user1.app.driver.wait(until.elementIsVisible(elem))
+    const text = await elem.getText()
+    expect(text).toEqual(users.owner.messages[0])
+
+    const fullMessages = await generalChannelUser1.getUserMessagesFull(users.owner.username)
+    const img = await fullMessages[1].findElement(By.tagName('img'))
+    await users.user1.app.driver.wait(until.elementIsVisible(img))
+    const imgSrc = await img.getAttribute('src')
+    expect(imgSrc).toEqual(expectedImgSrc)
+  })
+})
```

Comment:
I notice that all of our other multiple client e2e tests are part of a single test run where we test that a few different basic things are working, and that this is the only e2e test that only tests a single feature. 

What are the pros and cons of having a separate run vs. including this test as part of our main run?

For example, I suppose there is an upside to having this test be isolated from other tests, in that perhaps it protects it from giving a false negative.

On the other hand, having this test be part of the multiple clients test means that we are only doing the most time consuming and brittle piece of the process (invitation and initial connection over Tor) once.

@EmiM and @Kacper-RF what do you think?

Reply:
Yes exactly the most consuming thing beside whole stuff with building/setting environment on CI is joining to community.

In overall my vision for e2e tests is to include as many cases in one single test to save time when joining.
This has one drawback - maintaining and adding test cases to such a long scenario test is sometimes a bit of a pain.

I think we can watch how long it will take to pass all test files (including the one with the test user profile) on CI and decide if we should move those test cases to `multiple Clients.test.ts`